ostd/src/device/io_mem.rs

```diff
@@ -196,3 +196,10 @@ impl VmIoOnce for IoMem {
         self.writer().skip(offset + self.offset).write_once(new_val)
     }
 }
+
+impl Drop for IoMem {
+    fn drop(&mut self) {
+        // FIXME: After the PCI bus is removed from OSTD, we should uncomment the following code.
+        // super::dispatcher::io_mem::IO_MEM_DISPATCHER.recycle(self.pa..(self.pa + self.limit));
```

Comment:
Oh, I forgot to mention that this is actually not possible. 

Take a look at the `IoMem::slice` method. We need to remove this method before adding the `Drop` implementation for `IoMem'.

Reply:
I add the implementation directly without noticing the `IoMem::slice` method. With this method, I think the `Drop` implementation should be removed in this PR, as we don't have the necessary to recycle the `IoMem` for now. 

Additionally, deleting `IoMem::slice` may require more effort than we are willing to.

Reply:
Besides, `#[derive(Clone)]` should also be removed if the `Drop` implementation is present.

> Additionally, deleting `IoMem::slice` may require more effort than we are willing to.

I think we need to clarify a design question: Do multiple `IoMem`s guarantee that they will not intersect?
 - If the answer is yes, then `#[dervie(Clone)]` and `IoMem::slice` clearly violate the guarantee.
 - If the answer is no, then the backend of the `IoMemDispatcher` does not need to be an _allocator_. It can be as simple as checking if the area is valid and then creating the `IoMem`.
 - Otherwise, do you mean the answer should be yes, but you want to defer the work as future work?

Reply:
Yes, I think that the multiple `IoMem` instances should not overlap, but I would like to categorize this as future work.

Device drivers use the `Clone` and `IoMem::slice` methods to develop quickly, but this results in technical debt. Refactoring this code deserves a separate PR.

Reply:
Yes, of course we can leave that for future work.

Then I suggest we can add a FIXME/TODO somewhere to note the fact, as the current implementation actually does something undesirable and inconsistent (from the `IoMem` perspective and from the `IoMemDispatcher` perspective).